### ISSUE TITLE
Does Knowledge Distillation work with tasks which involve loss function as MSE??

### ISSUE THREAD

**Issue by rushi-the-neural-arch** [nathanhubens/fasterai1#2]

**Comment by nathanhubens** [comment]
Hi there ! 

Current implementation doesn't allow for other use cases than classification. However, I think that KD can still work for regression cases using the same idea. 

[This paper](http://www.cs.ox.ac.uk/files/11078/ICCV19_Distilling_Knowledge_From_a_Deep_Pose_Regressor_Network.pdf) describes how they use a MSE loss to do KD.

From what they say, your loss should be something like:
```python
KD_loss = α * MSE(student_output, teacher_output) +(1-α)* MSE(student_output, groundtruth)
```
 with α being a balancing term.

In this case, I'm not sure what the purpose of the teacher is except maybe adding some noise (=regularization).

**Comment by rushi-the-neural-arch** [comment]
Ohkay got your point. Thanks for the help @nathanhubens :)
